packages/core/src/App/Containers/RealAccountSignup/account-wizard.jsx

```diff
@@ -142,9 +144,16 @@ const AccountWizard = observer(props => {
             }
             return previous_state;
         });
+        setTempFormikData(previous_state => {
+            if (!previous_state.length) {
+                return getItems(get_items_props);
+            }
+            return previous_state;
+        });
```

Comment:
I'm curious how are we accessing the `previous_state` in `setTempFormikData` as I see it's not returning anything

Reply:
@shahzaib-deriv we can pass value, expression or callback in dispatch function. For example, you have this code:

const [counter, setCounter] = useState(0);

In this case you can set a new value these ways:

setCounter(1);
setCounter(counter + 1)
setCounter(previous_value => previous_value + 1)

Here is case #3. Supposed mobx action functions are like the dispatch function... I'll recheck it soon and come back

Reply:
True that we can pass a callback function to a setter in case of `useState` hook yet we're not defining any return from our function which should be giving us the whole state or we're not expecting any function as a parameter in `setTempFormikData` to be called

Reply:
@shahzaib-deriv It works for me, but I didn't find any confirmation to use it the same way as dispatch, so I've changed it. Thank you